frontend/components/RewardsHistory/RewardsHistory.tsx

```diff
@@ -0,0 +1,225 @@
+  return new Date(dateInMs * 1000).toLocaleDateString('en-US', {
+    month: 'short',
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+    hour12: true,
+    timeZone: 'UTC',
+  });
+};
+
+const RewardsHistoryTitle = () => <CardTitle title="Staking rewards history" />;
+
+const EarnedTag = () => (
+  <Tag color="success" className="m-0">
+    Earned
+  </Tag>
+);
+
+const NotEarnedTag = () => (
+  <Tag color="red" className="m-0">
+    Not earned
+  </Tag>
+);
+
+const Container = ({ children }: { children: ReactNode }) => (
+  <Flex
+    vertical
+    gap={24}
+    align="center"
+    justify="center"
+    style={{ height: MIN_HEIGHT }}
+  >
+    {children}
+  </Flex>
+);
+
+const Loading = () => (
+  <Container>
+    <Spin />
+  </Container>
+);
+
+const NoRewardsHistory = () => (
+  <Container>
+    <HistoryOutlined style={iconStyle} />
+    <Text type="secondary">There’s no history of rewards yet</Text>
+  </Container>
+);
+
+const ErrorLoadingHistory = ({ refetch }: { refetch: () => void }) => (
+  <Container>
+    <ApiOutlined style={iconStyle} />
+    <Text type="secondary">Error loading data</Text>
+    <Button onClick={refetch}>Try again</Button>
+  </Container>
+);
+
+const EpochTime = ({ epoch }: { epoch: EpochDetails }) => {
+  const timePeriod = useMemo(() => {
+    if (epoch.epochStartTimeStamp && epoch.epochEndTimeStamp) {
+      return `${getFormattedDate(epoch.epochStartTimeStamp)} - ${getFormattedDate(epoch.epochEndTimeStamp)} (UTC)`;
+    }
+    if (epoch.epochStartTimeStamp) {
+      return `${getEpochEndDate(epoch.epochStartTimeStamp)} (UTC)`;
+    }
+    return 'NA';
+  }, [epoch]);
+
+  return (
+    <Text type="secondary">
+      {getEpochEndDate(epoch.epochEndTimeStamp)}
+      &nbsp;
+      <Popover
+        arrow={false}
+        placement="topRight"
+        content={
+          <Flex vertical gap={4} className="text-sm" style={{ width: 280 }}>
+            <Title level={5} className="text-sm m-0">
+              Epoch duration
+            </Title>
+            <Text type="secondary" className="text-sm m-0">
+              {timePeriod}
+            </Text>
+            <a
+              href={`https://gnosisscan.io/tx/${epoch.transactionHash}`}
+              target="_blank"
+            >
+              End of epoch transaction {UNICODE_SYMBOLS.EXTERNAL_LINK}
+            </a>
+          </Flex>
+        }
+      >
+        <InfoCircleOutlined />
+      </Popover>
+    </Text>
+  );
+};
+
+const RewardsHistoryList = ({ reward }: { reward: StakingReward }) => (
+  <Flex vertical>
+    <ContractName>
+      <Text strong>{reward.name}</Text>
+    </ContractName>
+
+    {reward.history.map((epoch, index) => {
+      const currentEpochReward = epoch.reward
+        ? `~${balanceFormat(epoch.reward, 2)} OLAS`
+        : 'NA';
+
+      return (
+        <EpochRow
+          key={epoch.epochEndTimeStamp}
+          className={index === reward.history.length - 1 ? 'mb-16' : ''}
```

Comment:
can just be a gap={16} in Flex? I see `ContractName` has also bottom offset 16px